test/libsolidity/semanticTests/multiSource/free_function_transitive_import.sol

```diff
@@ -0,0 +1,27 @@
+==== Source: s1.sol ====
+function f() pure returns (uint) { return 1337; }
+contract C {
+  function g() public pure returns (uint) {
+    return f();
+  }
+}
+==== Source: s2.sol ====
+import "s1.sol";
+contract D is C {
+  function h() public pure returns (uint) {
+    return g();
+  }
+}
+==== Source: s3.sol ====
+import "s2.sol";
+import {f as f} from "s2.sol";
+contract E is D {
+  function i() public pure returns (uint) {
+    return f();
+  }
+}
+
+// ====
+// compileViaYul: also
+// ----
+// i() -> 1337
```

Comment:
Why don't you call `h`?

Reply:
Because `D` does not mark `h` as virtual. I think we are testing importing an imported symbol here.